### PR TITLE
feat: Completely decouple `rpc-types` to standalone crate

### PR DESCRIPTION
**Pull request by Evalir**
Makes `rpc-types` not depend on `reth-primitives`, which allows us to yank the crate into `alloy-rs/next`.

Compat functions were be added on `rpc-types-compat`.

### CODE REVIEW

File: crates/payload/builder/src/payload.rs [hunk 1]
```diff
@@ -111,7 +111,7 @@ impl From<BuiltPayload> for ExecutionPayloadEnvelopeV3 {
             // Spec:
             // <https://github.com/ethereum/execution-apis/blob/fe8e13c288c592ec154ce25c534e26cb7ce0530d/src/engine/cancun.md#specification-2>
             should_override_builder: false,
-            blobs_bundle: sidecars.into(),
+            blobs_bundle: sidecars.into_iter().map(|s| from_primitive_sidecar(s)).collect::<Vec<_>>().into(),
```
Comment (mattsse) [inline crates/payload/builder/src/payload.rs:114]:
```suggestion
            blobs_bundle: sidecars.into_iter().map(from_primitive_sidecar).collect::<Vec<_>>().into(),
```

File: crates/rpc/rpc-types-compat/src/block.rs [hunk 2]
```diff
@@ -94,8 +156,8 @@ fn from_block_with_transactions(
     transactions: BlockTransactions,
 ) -> Block {
     let uncles = block.ommers.into_iter().map(|h| h.hash_slow()).collect();
-    let header = Header::from_primitive_with_hash(block.header.seal(block_hash));
-    let withdrawals = if header.withdrawals_root.is_some() { block.withdrawals } else { None };
+    let header = from_primitive_with_hash(block.header.seal(block_hash));
+    let withdrawals = if header.withdrawals_root.is_some() { block.withdrawals.map(|withdrawals| withdrawals.into_iter().map(|w| from_primitive_withdrawal(w)).collect()) } else { None };
```
Comment (mattsse) [inline crates/rpc/rpc-types-compat/src/block.rs:160]:
```suggestion
    let withdrawals = if header.withdrawals_root.is_some() { block.withdrawals.map(|withdrawals| withdrawals.into_iter().map( from_primitive_withdrawal).collect()) } else { None };
```

File: crates/rpc/rpc-types-compat/src/engine/payload.rs [hunk 3]
```diff
@@ -360,6 +360,44 @@ pub fn convert_to_payload_body_v1(value: Block) -> ExecutionPayloadBodyV1 {
...
 }
 
+/// Transforms a [reth_primitives::BlobTransactionSidecar] into a [reth_rpc_types::BlobTransactionSidecar]
+pub fn from_primitive_sidecar(sidecar: reth_primitives::BlobTransactionSidecar) -> reth_rpc_types::BlobTransactionSidecar {
+    reth_rpc_types::BlobTransactionSidecar {
+        blobs: sidecar.blobs,
+        commitments: sidecar.commitments, 
+        proofs: sidecar.proofs,
+    }
+}
```
Comment (mattsse) [inline crates/rpc/rpc-types-compat/src/engine/payload.rs:370]: why do we have this twice?

File: crates/rpc/rpc-types/Cargo.toml [hunk 4]
```diff
@@ -25,10 +25,20 @@ serde_with = "3.3"
 serde_json.workspace = true
 jsonrpsee-types = { workspace = true, optional = true }
 alloy-primitives = { workspace = true, features = ["rand", "rlp"] }
+c-kzg = { workspace = true, features = ["serde"] }
+arbitrary = { workspace = true, features = ["derive"], optional = true }
+proptest = { workspace = true, optional = true }
+proptest-derive = { workspace = true, optional = true }
```
Comment (mattsse) [inline crates/rpc/rpc-types/Cargo.toml:31]: do we need these?

File: crates/rpc/rpc-types/Cargo.toml [hunk 5]
```diff
@@ -25,10 +25,20 @@ serde_with = "3.3"
 serde_json.workspace = true
 jsonrpsee-types = { workspace = true, optional = true }
 alloy-primitives = { workspace = true, features = ["rand", "rlp"] }
+c-kzg = { workspace = true, features = ["serde"] }
+arbitrary = { workspace = true, features = ["derive"], optional = true }
```
Comment (mattsse) [inline crates/rpc/rpc-types/Cargo.toml:29]: do we need arbitrary?
for what?

File: crates/rpc/rpc-types/src/net.rs [hunk 6]
```diff
@@ -0,0 +1,363 @@
...
+/// Ethereum Foundation Go Bootnodes
+pub static MAINNET_BOOTNODES : [&str; 4] = [
+    "enode://d860a01f9722d78051619d1e2351aba3f43f943f6f00718d1b9baa4101932a1f5011f16bb2b1bb35db20d6fe28fa0bf09636d26a87d31de9ec6203eeedb1f666@18.138.108.67:30303",   // bootnode-aws-ap-southeast-1-001
+    "enode://22a8232c3abc76a16ae9d6c3b164f98775fe226f0917b0ca871128a74a8e9630b458460865bab457221f1d448dd9791d24c4e5d88786180ac185df813a68d4de@3.209.45.79:30303",     // bootnode-aws-us-east-1-001
+    "enode://2b252ab6a1d0f971d9722cb839a42cb81db019ba44c08754628ab4a823487071b5695317c8ccd085219c3a03af063495b2f1da8d18218da2d6a82981b45e6ffc@65.108.70.101:30303",   // bootnode-hetzner-hel
+    "enode://4aeb4ab6c14b23e2c4cfdce879c04b0748a20d8e9b59e25ded2a08143e265c6c25936e74cbc8e641e3312ca288673d91f2f93f8e277de3cfa444ecdaaf982052@157.90.35.166:30303",   // bootnode-hetzner-fsn
+];
+
+/// Ethereum Foundation SEPOLIA BOOTNODES
+pub static SEPOLIA_BOOTNODES : [&str; 5] = [
```
Comment (mattsse) [inline crates/rpc/rpc-types/src/net.rs:128]: this does not belong here, please move back

File: crates/rpc/rpc-types/src/peer.rs [hunk 7]
```diff
@@ -0,0 +1,61 @@
...
+        Self(value.0, value.1)
+    }
+}
+
+impl<T> WithPeerId<T> {
+    /// Wraps the value with the peerid.
+    pub fn new(peer: PeerId, value: T) -> Self {
+        Self(peer, value)
+    }
+
```
Comment (mattsse) [inline crates/rpc/rpc-types/src/peer.rs:24]: this does not belong here please revert

**Review by mattsse — CHANGES_REQUESTED** [review]

File: crates/rpc/rpc-types/Cargo.toml (hunk 5 above)
Comment (Evalir) [reply crates/rpc/rpc-types/Cargo.toml]: removing—leftover from moving things around

File: crates/rpc/rpc-types/Cargo.toml (hunk 4 above)
Comment (Evalir) [reply crates/rpc/rpc-types/Cargo.toml]: also removed

File: crates/rpc/rpc-types-compat/src/engine/payload.rs (hunk 3 above)
Comment (Evalir) [reply crates/rpc/rpc-types-compat/src/engine/payload.rs]: There was already an old one elsewhere—removing

**Review by mattsse — COMMENTED** [review]
great this looks complete, I'll have a closer look in the am and will get this over the line

**Review by mattsse — CHANGES_REQUESTED** [review]
hmm,

the duplication of BlockId in primitives is actually a deal breaker for this.

we should inverse the dependency so that reth-primitives now depends on rpc-types, this allows us to get rid of some dupes,
BlockId,NodeRecord,serde-helpers

I'd also like to make serde-helpers a separate crate but we can do later

**Review by mattsse — APPROVED** [review]
lgtm,

we probably can do a few followups once we know what can be unified/simplified

**Pull request merged**